src/components/WindowSideBarButtons.js

```diff
@@ -12,78 +12,201 @@ import CanvasIndexIcon from './icons/CanvasIndexIcon';
  *
  */
 export class WindowSideBarButtons extends Component {
+  /**
+   * selects the given tab
+   * @param {*} tab the tab to activate
+   */
+  static activateTab(tab) {
+    tab.removeAttribute('tabindex');
+    tab.setAttribute('aria-selected', 'true');
+  }
+
   /** */
   constructor(props) {
     super(props);
+    const { windowId } = this.props;
     this.handleChange = this.handleChange.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.containerId = `${windowId}-sidebar-buttons`;
+    this.keys = {
+      end: 'End',
+      home: 'Home',
+      up: 'ArrowUp',
+      down: 'ArrowDown',
+    };
+
+    this.chars = {
+      end: 35,
+      home: 36,
+      up: 38,
+      down: 40,
+    };
   }
 
-  /** */
+  /**
+   *
+   */
+  componentDidMount() {
+    this.tabs = Array.from(document.querySelectorAll(`#${this.containerId} button[role=tab]`));
```

Comment:
I'm wondering if some of our testing woes may be coming from the use of `document.querySelectorAll` in the `componentDidMount` here.  I'm going to poke at if refs could be at all useful.